server/modules/refine.py

```diff
@@ -0,0 +1,74 @@
+import pandas as pd
+import numpy as np
+from .moduleimpl import ModuleImpl
+import json
+from .editcells import to_numeric
+from server.utils import safe_column_to_string
+import logging
+
+
+class Refine(ModuleImpl):
+
+    def render(wf_module, table):
+
+        edits_json = wf_module.get_param_raw('edits', 'string')
+        if edits_json == '':
+            return table
+
+        try:
```

Comment:
This is a very long block! Can you break it up into functions, perhaps on per if-clause?